### ISSUE TITLE
Scaffolding upgrade 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#349]
SoloSniper
I think dying scaffolding could be very useful..

Klaranth 
You mean it kills you when you use it?

**Comment by SwissalpS** [comment]
No, they want scaffolding to have a colour overlay so the same node can be used for marking out sections.

Currently we use ice and snow to do this, but avatar can't climb through those.

Alternative is to use geo caches, but they are opaque too.

Edit: I have also used mesecon wires and papyrus at times.

**Comment by Klaranth** [comment]
SoloSniper
yes, sorry, 'dyeing' :p the idea would be ease of building large projects that have multiple parts, and being able to distinguish them from eachother easier

FeXoR
(Some nodes can be "painted" with replaced if the colour is stored in param2 AFAIK)

SoloSniper
Replacer you mean?

FeXoR
Yea, right ;)

SoloSniper
And how would i do that ^.-

MCLV
I suspect you are going down the complete wrong rabbit hole with that one FeXoR

FeXoR
Set the replacer to the node of type and colour as you want to replace things with, turn of that param2 setting, replace the stuff you want to have of that material and colour. Hm, probably. What is the goal? :D

MCLV
The goal is to have colored scaffolding in world, I think

FeXoR
IMO that won't add much value and eat up preacious node space :/

SoloSniper
How do i 'turn off that param2 setting'? somewhat of a rookie to the technicals here

MCLV
What's a node limit and how do we break it?

FeXoR
32768. And why do you want to break stuff? x)

MCLV
To break through :D

FeXoR
Aux1 + Left
(The dropdown at the bottom)
(It's labeled "rotation" though. Maybe that actually won't work for colours after all?)

OgelGames 
We could have 255 colors of scaffolding with just one new node

SoloSniper
It works for some things

FeXoR
MCLV: If you mean inceasing the limit: I guess you can join the Mintest Engine team ;)

SoloSniper
not scaffolding though :/

OgelGames 
because scaffolding doesnt have a colorable paramtype2

SoloSniper
I see. Thanks!

OgelGames 
By the way, if you're wondering, we're at about 23k nodes currently, so we've used up over 2/3 of the nodes already

SoloSniper
Wait, sorry, what is this 'node count'?

FeXoR
The number of node type a server has (and the maximum it can have)

OgelGames 
Kinda crazy when you think about it, that we have that many unique nodes (plus another thousand or so items)

FeXoR
Yea, I only say: Bridger x)

SoloSniper
Yeah, that's an insane amount

FeXoR 
And of course the circular saw and the CNC

OgelGames 
thats also where scaffolding is from :P

FeXoR
Baked clay has colours and shapes. That alone is probably eating up about 1k node space.

Klaranth 
SoloSniper : Would the normal amount of colours for the scaffolding suffice, and NOT the whole gamut of all 256 colours?
If you want to use it as a way of setting up an example of a build, before using replacer to replace different coloured scaffodling with different nodes, is 15 colours sufficient? Or perhaps too much ?

FeXoR
Klaranth: See #offtopic: Most nodes come from shapes (moreblocks Circular Saw and Technic CNC) and colours - or both like Baked Clay. That is basically a single node in all shapes and colours being enough to get into the top 10 most node type mods x)

Then again 15 colours isn't that much compared to 48 shapes, hm.

Jeremy 
@Klaranth Yes ofc, even as much as 3 colors would work in my case 

Klaranth 
So, just red, blue, yellow for the colours, plus white and black for good measure ?
Gives a good 6 variations

SoloSniper
Hehe, yes that sounds splendid to me :)

FeXoR
(If we go for it I would go for the usual pallette)

**Comment by Klaranth** [comment]
SoloSniper
Idea for scaffolding coloring: the crafting should use 8 scaffolding and 1 dye to craft a batch.. sorta like baked clay works, it would make it less annoying

**Comment by SwissalpS** [comment]
as OgelGames mentioned, only one additional node is needed to add **up to** 255 more colours. One more node def no matter if 3, 7 or 255 colours.

**Comment by birdlover32767** [comment]
you can't actually have 255 extra colors with scaffolding since it uses "facedir" which limits it to 89 colors
plus, there are 8 different types of scaffolding so you have to add the color stuff 8 times (which is not hard i realize now)

anyway `unified_dyes` has a nice api for coloring nodes (see https://github.com/mt-mods/unifieddyes/blob/master/API.md) so this will be easier thanks to it

**Comment by SwissalpS** [comment]
89 colours are already plenty.

What are the 8 different scaffoldings? I only know and use one.

**Comment by birdlover32767** [comment]
> What are the 8 different scaffoldings? I only know and use one.

see https://bitbucket.org/Infinatum/scaffolding/src/930edb43d2460fb027255e7fbc17aa7b89532690/init.lua#lines-66
it defines 8 different scaffolding types

- wood
- reinforced wood
- wood platform
- reinforced wood platform
- iron
- reinforced iron
- iron platform
- reinforced iron platform

reinforced ones don't break if they float, and iron scaffolding requires a pick to mine

**Comment by SwissalpS** [comment]
on pandorabox only ``bridger:scaffolding`` is available and so I was convinced we were talking about that one.

**Comment by birdlover32767** [comment]
well there is only one bridger scaffolding, i thought the `scaffolding` mod was installed, but nope

so the bridger scaffolding code is:
```
minetest.register_node("bridger:scaffolding", {
	description = "Scaffolding",
	drawtype = "glasslike_framed_optional",
	tiles = {"bridges_scaffolding.png", "bridges_scaffolding_detail.png"},
	paramtype = "light",
	sunlight_propagates = true,
	walkable = false,
	climbable = true,
	groups = {cracky = 3, oddly_breakable_by_hand = 3, dig_immediate = 3},
	sounds = default.node_sound_wood_defaults(),
})
```

to make a colored version you can easily do this:
```
minetest.register_node("bridger:colored_scaffolding", {
	description = "Colored Scaffolding",
	drawtype = "glasslike_framed_optional",
	tiles = {"bridges_colored_scaffolding.png", "bridges_colored_scaffolding_detail.png"},
	paramtype = "light",
	sunlight_propagates = true,
	walkable = false,
	climbable = true,
	groups = {cracky = 3, oddly_breakable_by_hand = 3, dig_immediate = 3, ud_param2_colorable = 1}, -- ud_param2_colorable is new
	sounds = default.node_sound_wood_defaults(),
	airbrush_replacement_node = "bridger:scaffolding", -- new line
	on_dig = unifieddyes.on_dig -- new line
	paramtype2 = "color", -- new line
	palette = "unifieddyes_palette_extended.png", -- new line
})
```
the colored texture is just desaturated, maybe color-corrected scaffolding

**Comment by SwissalpS** [comment]
last commit was three years ago, PRs stacking up (https://github.com/v-rob/bridger/pulls) ... might have to fork or do an override in [pandorabox_custom].
Latter looks simpler.

**Comment by birdlover32767** [comment]
well, adding to `pandorabox_custom` is probably better (since it is custom to pandorabox after all)

but sure maybe open a pr if you wanna
